src/modules/reference/hooks.js

```diff
@@ -12,3 +13,49 @@ export const useReferenceGenomes = () => {
     }, [dispatch, referenceService]);
     return useSelector((state) => state.referenceGenomes);
 };
+
+
+/**
+ * @param {string | undefined} referenceGenomeID
+ * @param {string | null | undefined} nameQuery
+ */
+export const useGeneNameSearch = (referenceGenomeID, nameQuery) => {
+    const referenceService = useService("reference");
+
+    const [hasAttempted, setHasAttempted] = useState(false);
+    const [isFetching, setIsFetching] = useState(false);
+    const [data, setData] = useState([]);
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+        if (!referenceService || !referenceGenomeID || !nameQuery) return;
+
+        const params = new URLSearchParams({ name: nameQuery, name_fzy: "true", limit: "10" });
+        const searchUrl = `${referenceService.url}/genomes/${referenceGenomeID}/features?${params.toString()}`;
+
+        setError(null);
+
+        (async () => {
+            setIsFetching(true);
+
+            try {
+                const res = await fetch(searchUrl, { headers: { Accept: "application/json" } });
```

Comment:
Are we sure the reference service will always be public? If not, we should pass the auth headers as well.